app/src/main/kotlin/com/wire/android/ui/calling/controlButtons/MicrophoneButton.kt

```diff
@@ -5,26 +5,30 @@ import androidx.compose.foundation.layout.width
 import androidx.compose.material3.IconButton
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
-import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import com.wire.android.R
+import com.wire.android.ui.calling.OngoingCallState
 import com.wire.android.ui.theme.wireDimensions
 
 @Composable
-fun MicrophoneButton() {
-    var isCallMuted by remember { mutableStateOf(true) }
+fun MicrophoneButton(
+    ongoingCallState: OngoingCallState,
+    onMuteCall: () -> Unit,
+    onUnMuteCall: () -> Unit
+) {
     IconButton(
         modifier = Modifier.width(MaterialTheme.wireDimensions.defaultCallingControlsSize),
-        onClick = { isCallMuted = !isCallMuted }
+        onClick = {
+            if (ongoingCallState.isMuted)
+                onUnMuteCall()
+            else onMuteCall()
+        }
```

Comment:
I think this is something that belongs to the viewmodel, you could pass here muted state as a boolean value and handle the changing of the boolean state inside the viewmodel